### DISCUSSION TITLE
get $casts in model boot

### DISCUSSION THREAD

**Discussion by m-derakhshi** [laravel/framework#46918] [Q&A]
hi 
my model trait:[HasLoggableTrait] have boot like this:

``` 
protected static function bootHasLoggableTrait(): void
{
     print_r( $this->casts );
    $this->casts['created_at'] = ['datetime'];
}
```

Because the function is static, the information cannot be accessed and changed.

**Comment by Sophist-UK** [comment]
Yes - a `static` function cannot have access to the data of a specific instance - it can only operate on the parameters explicitly passed to it and return an object.

Remove `static` to make this work.
